colour/recovery/tests/test_jakob2019.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+"""
+Defines unit tests for :mod:`colour.recovery.jakob2019` module.
+"""
+
+from __future__ import division, unicode_literals
+
+import numpy as np
+import unittest
+
+from colour import (SpectralDistribution, COLOURCHECKER_SDS, ILLUMINANT_SDS,
+                    ILLUMINANTS, STANDARD_OBSERVER_CMFS,)
+from colour.colorimetry import sd_to_XYZ
+from colour.difference import delta_E_CIE1976
+from colour.models import (RGB_COLOURSPACES, XYZ_to_RGB,
+                           XYZ_to_Lab)
+from colour.recovery.jakob2019 import (RGB_to_sd_Jakob2019, spectral_model,
+                                       error_function,
+                                       dimensionalise_coefficients,
+                                       DEFAULT_SPECTRAL_SHAPE_JAKOB_2019,
+                                       ACCEPTABLE_DELTA_E)
+
+
+__author__ = 'Colour Developers'
+__copyright__ = 'Copyright (C) 2013-2020 - Colour Developers'
+__license__ = 'New BSD License - https://opensource.org/licenses/BSD-3-Clause'
+__maintainer__ = 'Colour Developers'
+__email__ = 'colour-developers@colour-science.org'
+__status__ = 'Production'
+
+__all__ = ['TestRGB_to_sd_Jakob2019']
+
+cmfs = STANDARD_OBSERVER_CMFS['CIE 1931 2 Degree Standard Observer']
+colourspace = RGB_COLOURSPACES["ProPhoto RGB"]
+D65 = SpectralDistribution(ILLUMINANT_SDS["D65"])
+D65_xy = ILLUMINANTS["CIE 1931 2 Degree Standard Observer"]["D65"]
+
+
+class TestRGB_to_sd_Jakob2019(unittest.TestCase):
+    """
+    Defines :func:`colour.recovery.jakob2019.RGB_to_sd_Jakob2019`
+    definition unit tests methods.
+    """
+
+    def test_roundtrip_colorchecker(self):
+        """
+        Tests :func:`colour.recovery.jakob2019.RGB_to_sd_Jakob2019` definition
+        round-trip errors using a color checker.
+        """
+
+        for name, sd in COLOURCHECKER_SDS['ColorChecker N Ohta'].items():
+            # The colours aren't too saturated and the tests should pass with
+            # or without feedback.
+            for use_feedback in [False, True]:
+                XYZ = sd_to_XYZ(sd, illuminant=D65) / 100
+
+                RGB = XYZ_to_RGB(
+                    XYZ,
+                    D65_xy,
+                    colourspace.whitepoint,
+                    colourspace.XYZ_to_RGB_matrix,
+                )
+
+                recovered_sd, error = RGB_to_sd_Jakob2019(
```

Comment:
Codacy found an issue: [Unused variable 'recovered_sd'](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5753724)

Reply:
This is actually not me writing that but Codacy.